client/src/components/Viewer/Viewer.js

```diff
@@ -55,6 +55,9 @@ function Viewer(props) {
         setIsConnected(true);
         socket.emit('addUser', {
           token: JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
+          user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user'][
+            'user_email'
+          ],
```

Comment:
The extraction of `user_email` from the `dfs-user` local storage should be checked for null or undefined values to avoid potential runtime errors. Consider adding a fallback or error handling mechanism if the `user_email` is not present.
```diff
-           user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user'][
+           user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user']?.['user_email'] || 'default_user_email',
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*